library/core/src/ops/index_range.rs

```diff
@@ -132,10 +133,9 @@ impl Iterator for IndexRange {
     }
 
     #[inline]
-    fn advance_by(&mut self, n: usize) -> Result<(), usize> {
-        let original_len = self.len();
-        self.take_prefix(n);
-        if n > original_len { Err(original_len) } else { Ok(()) }
+    fn advance_by(&mut self, n: usize) -> Result<(), NonZeroUsize> {
+        let taken = self.take_prefix(n);
```

Comment:
`IndexRange::take_prefix` exists to help implement `advance_by`, so feel free to change its signature if it would help these.